core/src/main/java/tech/tablesaw/io/fixed/FixedWidthReader.java

```diff
@@ -0,0 +1,620 @@
+            Table table = Table.create(options.tableName());
+
+            String[] headerNames = getHeaderNames(options, types, parser);
+            if (headerNames == null) {
+                return table;
+            }
+            List<String> headerRow = Lists.newArrayList(headerNames);
+
+            String[] columnNames = selectColumnNames(headerRow, types);
+
+            cleanNames(headerRow);
+            for (int x = 0; x < types.length; x++) {
+                if (types[x] != SKIP) {
+                    String columnName = headerRow.get(x);
+                    if (Strings.isNullOrEmpty(columnName)) {
+                        columnName = "Column " + table.columnCount();
+                    }
+                    Column<?> newColumn = types[x].create(columnName);
+                    table.addColumns(newColumn);
+                }
+            }
+            int[] columnIndexes = new int[columnNames.length];
+            for (int i = 0; i < columnIndexes.length; i++) {
+                // get the index in the original table, which includes skipped fields
+                columnIndexes[i] = headerRow.indexOf(columnNames[i]);
+            }
+
+            addRows(options, types, parser, table, columnNames, columnIndexes);
+            return table;
+        } finally {
+            if (options.reader() == null) {
+                // if we get a reader back from options it means the client opened it, so let the client close it
+                // if it's null, we close it here.
+                parser.stopParsing();
+                reader.close();
+            }
+        }
+    }
+
+    private String[] getHeaderNames(FixedWidthReadOptions options, ColumnType[] types, FixedWidthParser parser) {
+        String[] headerNames;
+        if (options.header()) {
+            headerNames = parser.parseNext();
+            // work around issue where Univocity returns null if a column has no header.
+            for (int i = 0; i < headerNames.length; i++) {
+                if (headerNames[i] == null) {
+                    headerNames[i] = "C" + i;
+                }
+            }
+        } else {
+            headerNames = makeColumnNames(types);
+        }
+        return headerNames;
+    }
+
+    private Reader getReader(FixedWidthReadOptions options, byte[] bytes)
+            throws FileNotFoundException {
+        if (bytes != null) {
+            return new InputStreamReader(new ByteArrayInputStream(bytes));
+        }
+
+        if (options.inputStream() != null) {
+            return new InputStreamReader(options.inputStream());
+        } else if (options.reader() != null) {
+            return options.reader();
+        }
+        return new InputStreamReader(new FileInputStream(options.file()));
+    }
+
+    /**
+     * Returns column types for this table as an array, the types are either provided in read options, or calculated by
+     * scanning the data
+     *
+     * @throws IOException
```

Comment:
Can you remove this? It doesn't add anything. Normally you add this to explain when an exception is thrown